Makefile

```diff
@@ -28,7 +28,7 @@ ifdef DEBUG
   CCFLAGS += -ggdb -O0
   LDFLAGS += -ggdb
 else
-  CCFLAGS += -Os
+  CCFLAGS += -O2 
```

Comment:
> This change is huge 

It is a slight increase in generated flash size, but I think that you mean that the number of non-aligned fetch exceptions is dramatically less.  Wouldn't it be better to disable this one specific size optimatisation since there are other differences between these two options.